packages/x-charts/src/internals/index.ts

```diff
@@ -1 +1,26 @@
+// Components
+export * from './components/ChartsAxesGradients';
+
+// hooks
+export { useReducedMotion } from '../hooks/useReducedMotion';
+export { useSeries } from '../hooks/useSeries';
+
+// utils
+export * from './defaultizeValueFormatter';
 export * from './configInit';
+
+// contexts
+
+export * from '../context/CartesianContextProvider';
+export * from '../context/DrawingProvider';
+export * from '../context/InteractionProvider';
+export * from '../context/SeriesContextProvider';
+export * from '../context/ZAxisContextProvider';
+
+// series configuration
+export * from '../models/seriesType/config';
+export * from '../models/seriesType/common';
+
+export * from '../models/helpers';
+export * from '../models/z-axis';
+export * from '../models/axis';
```

Comment:
This file will probably cause a lot of unnecessary cyclic deps 😓

Reply:
Why do you think so?

The `internal/index.ts` is exporting all the elements from the community package we need for creating the pro one but don't want to expose to the users.

- they are not documented in API pages
- they can only be imported with `from '@mui/x-charts/internal` which is assumed to be enough to know those API are to be considered unstable

So the community package has 3 kinds of elements:
- Those publically exported by `@mui/x-charts` and `@mui/x-charts/Xxxx` (in blue)
- Those privately exported by `@mui/x-charts/internals` (in red)
- Those not exported (in yellow)

![image](https://github.com/mui/mui-x/assets/45398769/fbf1c5db-fbb1-45b1-beff-b78e989db92a)

Since import/export is only going from MIT to pro, this should not create any loop. But I might miss something

Reply:
Internals itself is a folder with content, and it is exporting files and folders outside of itself, on the same level.

Eg: `export * from '../context/CartesianContextProvider';`

The `../context` is at the same level as `../internals`

If on the `CartesianContextProvider.tsx` file, we remove `import { SeriesId } from '../models/seriesType/common';` and reimport it, it will prefer the to import from `../internals` and will cause a cycle.

Reply:
By convention, we never import from `/internals`. You can import from `internals/...` but not the index file. This index file is only here to be used by the pro packages. THis convention should prevent that kind of issue

Reply:
Do you think it would be useful to have an eslint rule about it then?

Reply:
I never got any issues with that in 3 years 😇

Reply:
🤔 my auto import creates cyclic deps instantly with these changes, then I have to manually change the import. But we can look at that in another MR